x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/trigger/ScheduleTriggerEngineMock.java

```diff
@@ -50,37 +53,42 @@ public ScheduleTriggerEvent parseTriggerEvent(TriggerService service, String wat
 
     @Override
     public void start(Collection<Watch> jobs) {
```

Comment:
this method by itself is not thread safe. I  would advise to synchronize this method, or use a shared lock between this start and stop, which negates the need for the atomic reference, or throw an exception if attempted to start/stop twice.

Reply:
Good point, I'll take another look at this. There's a few cases where this could potentially have issues now that I look at it again.

Reply:
This method is invoked from `TriggerService#start(...)` and that method is `synchronized`, so I don't think we need to `synchronize` this method.

Reply:
I've added `synchronized` to the methods which modify `watches` to ensure this is thread safe. How does this look @jakelandis?